### PR TITLE
feat/add-nextjs-demo-app

### PR DESCRIPTION
**Pull request by abbaseya**

### CODE REVIEW

**Review by JosephSamirL — APPROVED** [review]

File: packages/demo-nextjs/.next/BUILD_ID
Comment (abbaseya) [inline packages/demo-nextjs/.next/BUILD_ID:1]: @JosephSamirL this whole folder `.next` should be ignored in git!

File: packages/demo-nextjs/middleware.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+// middleware.js
+
+import { NextResponse } from 'next/server';
+import sdkInstance from '@/lib/convertSdk';
```
Comment (abbaseya) [inline packages/demo-nextjs/middleware.js:4]: @JosephSamirL I cannot find `lib/convertSdk` anywhere, I assume that's where the ConvertSDk is being initialized with our SDK key. Most probably the `lib` folder is ignored in git (_there's a reason for that_). I think you should rename `lib/convertSdk` to something like `convert/init`

**Review by abbaseya — COMMENTED** [review]
@JosephSamirL added a couple of comments, you also need to update the main `README.md` to include this demo

File: packages/demo-nextjs/middleware.js (hunk 1 above)
Comment (JosephSamirL) [reply packages/demo-nextjs/middleware.js:4]: Changed to be under file convert because there were two files ignored convertSdk and datastore

**Pull request closed without merging**